internal/graph/check.go

```diff
@@ -136,22 +131,41 @@ func WithMaxConcurrentReads(limit uint32) LocalCheckerOption {
 	}
 }
 
+func WithDelegate(delegate CheckResolver) LocalCheckerOption {
+	return func(d *LocalChecker) {
+		d.delegate = delegate
+	}
+}
+
+func WithCachedResolver(opts ...CachedCheckResolverOpt) LocalCheckerOption {
+	return func(d *LocalChecker) {
+		cachedCheckResolver := NewCachedCheckResolver(
+			d,
+			opts...,
+		)
+		d.SetDelegate(cachedCheckResolver)
+	}
+}
+
 // NewLocalChecker constructs a LocalChecker that can be used to evaluate a Check
 // request locally.
-func NewLocalChecker(ds storage.RelationshipTupleReader, opts ...LocalCheckerOption) *LocalChecker {
+func NewLocalChecker(ds storage.RelationshipTupleReader, opts ...LocalCheckerOption) CheckResolver {
 	checker := &LocalChecker{
 		ds:                 ds,
 		concurrencyLimit:   defaultResolveNodeBreadthLimit,
 		maxConcurrentReads: defaultMaxConcurrentReadsForCheck,
 	}
+	checker.delegate = checker // by default, a LocalChecker delegates/dispatchs subproblems to itself (e.g. local dispatch) unless otherwise configured.
```

Comment:
https://github.com/openfga/openfga/pull/476#discussion_r1089530447

I'm sorry I keep bringing this up but I still hold the opinion that this line is odd. LocalChecker has a `delegate` which can hold a CachedLocalChecker. Meanwhile, CachedLocalChecker has also a `delegate` which holds a LocalChecker. It's like the [Spiderman meme](https://knowyourmeme.com/memes/spider-man-pointing-at-spider-man)..

I won't block the PR though, I'll think if there is a better way of doing this and if there is i'll send a PR.

Reply:
Let's have a follow up PR if there is a better way later.